drivers/input/input_sbus.c

```diff
@@ -0,0 +1,364 @@
+/*
+ * Copyright (c) 2024 CogniPilot Foundation
+ * Copyright (c) 2024 NXP Semiconductors
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT futaba_sbus
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/input/input.h>
+#include <zephyr/irq.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/sys/time_units.h>
+#include <zephyr/sys_clock.h>
+#include <zephyr/drivers/uart.h>
+
+LOG_MODULE_REGISTER(futaba_sbus, CONFIG_INPUT_LOG_LEVEL);
+
+/* Driver config */
+struct input_channel_config {
+	uint32_t channel;
+	uint32_t type;
+	uint32_t zephyr_code;
+};
+
+const struct uart_config uart_cfg_sbus = {.baudrate = 100000,
+					  .parity = UART_CFG_PARITY_EVEN,
+					  .stop_bits = UART_CFG_STOP_BITS_2,
+					  .data_bits = UART_CFG_DATA_BITS_8,
+					  .flow_ctrl = UART_CFG_FLOW_CTRL_NONE};
```

Comment:
```suggestion
const struct uart_config uart_cfg_sbus = {
        .baudrate = 100000,
        .parity = UART_CFG_PARITY_EVEN,
        .stop_bits = UART_CFG_STOP_BITS_2,
        .data_bits = UART_CFG_DATA_BITS_8,
        .flow_ctrl = UART_CFG_FLOW_CTRL_NONE,
};
```

Reply:
also should the baud rate come from devicetree?

Reply:
As far as I know SBUS has a fixed baudrate of 100000

Reply:
ok but in the example devicetree you set it on the parent device

```
&lpuart6 {
	current-speed = <100000>;
```

so should this code pick the number from devicetree? Or you could drop it from the example if it's hardcoded I guess, but I imagine it maybe convenient to allow changing it for testing purposes

Reply:
Removed the baudrate lines